frontend/fingertips-frontend/lib/search/searchService.ts

```diff
@@ -0,0 +1,62 @@
+import { IndicatorSearch, IndicatorSearchResult } from './searchResultData';
+import { AzureKeyCredential, SearchClient } from '@azure/search-documents';
+
+type Indicator = {
+  IID: string;
+  Descriptive: {
+    Name: string;
+    Definition: string;
+    DataSource: string;
+  };
+  DataChange: {
+    LastUploadedAt: string;
+  };
+};
+
+const getEnvironmentVariable = (name: string): string => {
+  return (
+    process.env[name] ??
+    (() => {
+      throw new Error(`Missing environment variable ${name}`);
+    })()
+  );
+};
+
+export class SearchService implements IndicatorSearch {
+  readonly searchClient: SearchClient<Indicator>;
+
+  constructor() {
+    const serviceName = getEnvironmentVariable('DHSC_AI_SEARCH_SERVICE_NAME');
+    const indexName = getEnvironmentVariable('DHSC_AI_SEARCH_INDEX_NAME');
+    const apiKey = getEnvironmentVariable('DHSC_AI_SEARCH_API_KEY');
+
+    this.searchClient = new SearchClient<Indicator>(
+      `https://${serviceName}.search.windows.net`,
+      indexName,
+      new AzureKeyCredential(apiKey)
+    );
+  }
+
+  async searchByIndicator(indicator: string): Promise<IndicatorSearchResult[]> {
+    const query = `/.*${indicator}.*/`;
+
+    const searchResponse = await this.searchClient.search(query, {
+      queryType: 'full',
```

Comment:
Do we need `queryType: 'full'` here? I _think_ I had to add that to allow fuzzy search to be enabled (which you're not using here).

Reply:
Needs to be `full` if you want to use Lucene query syntax, which I think includes wildcards (quick test seems to confirm this).

Reply:
I'm not sure you need the full Lucene syntax for wildcards, but I'm happy to leave it using this syntax for now.